pkgs/top-level/dhall-packages.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, newScope
+, overrides ? (self: super: {})
+}:
+
+let
+  packages = self:
+    let
+      callPackage = newScope self;
```

Comment:
Not a fan of the complexity introduced by this scoping override (not a fan of how `haskellPackages` overcomplicates many things in general). Can we keep it simple? `let` is a great idea.

Reply:
@Profpatsch: Actually, I was following how `pythonPackages` did things (which happens to be the same as `haskellPackages`)

I'm not exactly sure what alternative approach you are suggesting, but it sounds like you are recommending that I use `pkgs.callPackages` (with no scoping override).  I gave that a try and ran into the following issues:

* It's now more difficult for one Dhall package to depend upon another Dhall package

  ... since the non-overriden `callPackage` doesn't bring any Dhall packages in scope

* You lose the ability to override `buildDhallPackage`

  ... which is really useful, for the same reason that it's useful to override `pkgs.haskellPackage.mkDerivation`.  For example, you could globally override `buildDhallPackage` to set `source = true;` to build source distributions for Dhall packages

Or did I misunderstand what you are requesting?  Perhaps if you included a short code sample of what you had in mind it would be more clear

Reply:
I’m sorry, I was in a bad mood yesterday and did this review against my better judgement. I should do better than that.

We should use the approach in this PR because of two reasons:

1) It’s the default, with precedence set by `haskellPackages` and `pythonPackages`. There’s a lot of value in regularity, even if the usability or design is not the greatest.
2) There hasn’t been a better approach so far.